src/OMSimulatorLib/System.cpp

```diff
@@ -1087,7 +1087,7 @@ oms_status_enu_t oms::System::addConnection(const oms::ComRef& crefA, const oms:
   {
     // allow non-real connections to tables
   }
-  else if(conA->getType() == oms_signal_type_integer && conB->getType() == oms_signal_type_enum)
+  else if((conA->getType() == oms_signal_type_integer || conA->getType() == oms_signal_type_enum)  && (conB->getType() == oms_signal_type_enum || conB->getType() == oms_signal_type_integer))
```

Comment:
Shoudln't it be like this?
```else if((conA->getType() == oms_signal_type_integer && conB->getType() == oms_signal_type_enum) || (conA->getType() == oms_signal_type_enum && conB->getType() == oms_signal_type_integer))```

Reply:
this is fixed with  https://github.com/OpenModelica/OMSimulator/pull/1012/commits/bb23e8db430311948b76c3df81655b5436808a8d